examples/vision/learnable_resizer.py

```diff
@@ -0,0 +1,273 @@
+
+    # Skip connection
+    x = layers.Add()([bottleneck, x])
+
+    # Final resized image
+    x = layers.Conv2D(filters=3, kernel_size=7, strides=1, padding="same")(x)
+    final_resize = layers.Add()([naive_resize, x])
+
+    return final_resize
+
+
+"""
+## Visualize the outputs of the learnable resizing module
+
+Here, we visualize how the resized images would look like after being passed through the
+random weights of the resizer. 
+"""
+
+sample_images, _ = next(iter(train_ds))
+
+plt.figure(figsize=(10, 10))
+for i, image in enumerate(sample_images[:9]):
+    ax = plt.subplot(3, 3, i + 1)
+    image = tf.image.convert_image_dtype(image, tf.float32)
+    resized_image = learnable_resizer(image[None, ...])
+    plt.imshow(resized_image.numpy().squeeze())
+    plt.axis("off")
+
+"""
+## Model building utility
+"""
+
+
+def get_model():
+    backbone = tf.keras.applications.DenseNet121(
+        weights=None, include_top=True, classes=2
+    )
+    backbone.trainable = True
+
+    inputs = layers.Input((INP_DIM[0], INP_DIM[1], 3))
+    x = layers.experimental.preprocessing.Rescaling(scale=1.0 / 255)(inputs)
+    x = learnable_resizer(x)
+    outputs = backbone(x)
+
+    return tf.keras.Model(inputs, outputs)
+
+
+"""
+## Compile and train our model with learnable resizer
+"""
+
+model = get_model()
+model.compile(
+    loss=keras.losses.CategoricalCrossentropy(label_smoothing=0.1),
+    optimizer="sgd",
+    metrics=["accuracy"],
+)
+model.fit(train_ds, validation_data=validation_ds, epochs=EPOCHS)
+
+"""
+## Visualize the outputs of the trained visualizer
+"""
+
+learned_resizer = tf.keras.Model(model.input, model.layers[-2].output)
+
+plt.figure(figsize=(10, 10))
+for i, image in enumerate(sample_images[:9]):
+    ax = plt.subplot(3, 3, i + 1)
+    image = tf.image.convert_image_dtype(image, tf.float32)
+    resized_image = learned_resizer(image[None, ...])
+    plt.imshow(resized_image.numpy().squeeze())
+    plt.axis("off")
+
+"""
+As we can see the visuals of the images have improved with training. Additionally, you
+can find this repository that shows the benefits of using the resizing module. Below is a
+comparison:
+
+|           Model           	| Number of  parameters (Million) 	| Top-1 accuracy 	|
+|:-------------------------:	|:-------------------------------:	|:--------------:	|
+|   With learnable resizer  	|             7.051717            	|      52.02     	|
+| Without learnable resizer 	|             7.039554            	|      50.3      	|
+
+Note the above-reported models were trained for 10 epochs on 90% of the training set of
+Cats-vs-Dogs unlike this example. Also, note that the increase in the number of
+parameters due to the resizing module is very negligible. You can reproduce these results
+from [this repository](https://github.com/sayakpaul/Learnable-Image-Resizing). To ensure
+that the improvement in the performance is not due to stochasticity, the models were
+trained using the same initial random weights. 
```

Comment:
Suggestion: refactoring (with 80 char per line limit)

```suggestion
Note: Unlike this example, the comparable models in the above table were trained
for 10 epochs on 90% of the training set of the Cats-vs-Dogs dataset. Notice the
increase in the number of parameters—this is because the resizing module is
negligible. You can reproduce these results by using
[`sayakpaul/Learnable-Image-Resizing`](https://github.com/sayakpaul/Learnable-Image-Resizing)
repository. And, to make sure that the improvement in the performance is not due
to stochasticity, the models were trained using the same initial random weights.
```